Cargo.toml

```diff
@@ -31,6 +31,7 @@ sync = ["async-std-runtime"]
 bson-u2i = ["bson/u2i"]
 
 [dependencies]
+async-lock = "2.3.0"
```

Comment:
To avoid bringing in a new dependency, can we update this PR to use tokio's [`RwLock`](https://docs.rs/tokio/1.2.0/tokio/sync/struct.RwLock.html) here instead? We use it in a few places in the driver already.

Reply:
Thanks, that's a fair point!